app/code/Magento/InventorySales/Model/CheckItemsQuantity.php

```diff
@@ -67,11 +68,13 @@ public function execute(array $items, int $stockId) : void
     {
         $productTypes = $this->getProductTypesBySkus->execute(array_keys($items));
         foreach ($items as $sku => $qty) {
-            if (false === $this->isSourceItemsAllowedForProductType->execute($productTypes[$sku])) {
+            if (false === $this->isSourceItemsAllowedForProductType->execute($productTypes[$sku])
```

Comment:
`InventorySales` module *must not* have the dependency on `ConfigurableProduct` module